File: app/src/main/java/com/fonfon/noloss/ui/main/DevicesView.java
package com.fonfon.noloss.ui.main;

import com.fonfon.noloss.lib.ActivityEvent;
import com.fonfon.noloss.lib.Device;
import com.fonfon.noloss.viewstate.DevicesViewState;
import com.hannesdorfmann.mosby3.mvp.MvpView;

import io.reactivex.Observable;

public interface DevicesView extends MvpView {

  Observable<ActivityEvent> onLifecycleIntent();

  Observable<Object> newDeviceIntent();

  Observable<Device> deleteDeviceIntent();

  Observable<Device> alertDeviceIntent();

  Observable<Device> updateDeviceIntent();

  Observable<Object> refreshIntent();

  void render(DevicesViewState state);
}

File: app/src/main/java/com/fonfon/noloss/ui/main/DevicesView.kt
package com.fonfon.noloss.ui.main

import com.fonfon.noloss.DevicesViewState
import com.fonfon.noloss.lib.ActivityEvent
import com.fonfon.noloss.lib.Device
import com.hannesdorfmann.mosby3.mvp.MvpView
import io.reactivex.Observable

interface DevicesView : MvpView {

  fun onLifecycleIntent(): Observable<ActivityEvent>

  fun newDeviceIntent(): Observable<Any>

  fun deleteDeviceIntent(): Observable<Device>

  fun alertDeviceIntent(): Observable<Device>

  fun updateDeviceIntent(): Observable<Device>

  fun refreshIntent(): Observable<Any>

  fun render(state: DevicesViewState)
}
